### PR TITLE
feat(core/chart) enable native chart.js callbacks

### PR DESCRIPTION
**Pull request by peppedeka**

### CODE REVIEW

File: src/core/chart/chart.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@
  * If not, see http://www.gnu.org/licenses/.
  *
  */
-
+/* tslint:disable: no-eval */
```
Comment (trik) [inline src/core/chart/chart.ts:22]: please don't disable tslint rules if it's not really necessary. using eval brings a lot of security issues that we really want avoid. you can build the function at runtime from a string using the Function constructor

File: src/core/chart/chart.ts [hunk 2]
```diff
@@ -138,6 +138,88 @@ export class AjfChartComponent implements AfterViewInit, OnChanges {
 
   private _fixChartOptions(options: ChartOptions): ChartOptions {
     options = options || {};
+    if (options.legendCallback) {
```
Comment (trik) [inline src/core/chart/chart.ts:141]: there's no need to cast as any... you could replace with something like
```
  if (options.legendCallback && typeof options.legendCallback === 'string') {
      const legendCallback = options.legendCallback;
      options.legendCallback = new Function(legendCallback) as (chart: Chart) => string;
    }
```

File: src/core/reports/interface/widgets/widget.ts [hunk 3]
```diff
@@ -21,12 +21,16 @@
...
 import {AjfStyles} from '../styles';
+
 import {AjfWidgetType} from './widget-type';
 
 export interface AjfWidget {
   widgetType: AjfWidgetType;
   styles: AjfStyles;
   visibility: AjfCondition;
   repetitions?: AjfFormula;
+  options?: ChartOptions;
```
Comment (trik) [inline src/core/reports/interface/widgets/widget.ts:35]: why do we need to add the ChartOptions to the base widget interface? what's the reason to have some chart options for a layout widget?

Comment (peppedeka) [reply src/core/reports/interface/widgets/widget.ts:35]: for this reason => https://github.com/peppedeka/ajf/blob/enable-native-chartjs-callbacks/src/material/reports/chart-widget.html#L6

File: src/core/chart/chart.ts (hunk 2 above)
Comment (peppedeka) [reply src/core/chart/chart.ts:141]: if not cast as any  ==> 
image

Comment (trik) [reply src/core/chart/chart.ts:149]: Schermata 2020-06-12 alle 16 38 30
sorry, i don't get the point... i have no error in IDE and if can build the code successfully

File: src/core/reports/interface/widgets/widget.ts (hunk 3 above)
Comment (trik) [reply src/core/reports/interface/widgets/widget.ts:35]: so what? options for chart widget are already defined here..
https://github.com/gnucoop/ajf/blob/master/src/core/reports/interface/widgets/chart-widget.ts#L36

Comment (peppedeka) [reply src/core/reports/interface/widgets/widget.ts:35]: What?why ajf-chart take chartjs option from widget??????is code before my pull request

**Pull request merged**